source/setup/components/RemoteFileTree.js

```diff
@@ -231,7 +231,7 @@ class RemoteFileTree extends Component {
     render() {
         return (
             <Choose>
-                <When condition={this.props.rootDirectory && this.props.rootDirectory.directories.length > 0}>
+                <When condition={this.props.rootDirectory && this.props.rootDirectory.files.length > 0}>
```

Comment:
Maybe instead of replacing The condition, add the other one as well with ||? `this.prop.rootDirectory && (this.props.rootDirectory.directories.length > 0 || this.props.rootDirectory.files.length > 0)`? This way we can ensure we cover both no files and no directories.

Reply:
Excellent, i think i saw one prop for that in the code, let me check if i can use that or i need to add a new one.

Reply:
I've reused the prop, it looks like it was meant for the condition this PR is fixing but it was not used for some reason. Let me know what you guys think.
Cheers!